solidity/test/isms/ExternalBridgeTest.sol

```diff
@@ -0,0 +1,272 @@
+        testRecipient = new TestRecipient();
+        encodedMessage = _encodeTestMessage();
+        messageId = Message.id(encodedMessage);
+    }
+
+    /* ============ Hook.postDispatch ============ */
+
+    function test_postDispatch() public {
+        bytes memory encodedHookData = _encodeHookData(messageId);
+        originMailbox.updateLatestDispatchedId(messageId);
+        _expectOriginExternalBridgeCall(encodedHookData);
+
+        hook.postDispatch{value: GAS_QUOTE}(testMetadata, encodedMessage);
+    }
+
+    function test_postDispatch_revertWhen_chainIDNotSupported() public {
+        bytes memory message = MessageUtils.formatMessage(
+            0,
+            uint32(0),
+            DESTINATION_DOMAIN,
+            TypeCasts.addressToBytes32(address(this)),
+            3, // wrong domain
+            TypeCasts.addressToBytes32(address(testRecipient)),
+            testMessage
+        );
+
+        originMailbox.updateLatestDispatchedId(Message.id(message));
+        vm.expectRevert(
+            "AbstractMessageIdAuthHook: invalid destination domain"
+        );
+        hook.postDispatch(testMetadata, message);
+    }
+
+    function test_postDispatch_revertWhen_notLastDispatchedMessage() public {
+        vm.expectRevert(
+            "AbstractMessageIdAuthHook: message not latest dispatched"
+        );
+        hook.postDispatch(testMetadata, encodedMessage);
+    }
+
+    function test_postDispatch_revertWhen_tooMuchValue() public {
+        vm.deal(address(this), uint256(2 ** 255 + 1));
+        bytes memory excessValueMetadata = StandardHookMetadata
+            .overrideMsgValue(uint256(2 ** 255 + 1));
+
+        originMailbox.updateLatestDispatchedId(messageId);
+        vm.expectRevert(
+            "AbstractMessageIdAuthHook: msgValue must be less than 2 ** 255"
+        );
+        hook.postDispatch(excessValueMetadata, encodedMessage);
+    }
+
+    /* ============ ISM.verifyMessageId ============ */
+
+    function test_verifyMessageId_asyncCall() public {
+        bytes memory encodedHookData = abi.encodeCall(
+            AbstractMessageIdAuthorizedIsm.verifyMessageId,
+            (messageId)
+        );
+        _externalBridgeDestinationCall(encodedHookData, 0);
+
+        assertTrue(ism.isVerified(encodedMessage));
+    }
+
+    function test_verifyMessageId_externalBridgeCall() public virtual {
+        bytes memory externalCalldata = _encodeExternalDestinationBridgeCall(
+            address(hook),
+            address(ism),
+            0,
+            messageId
+        );
+
+        ism.verify(externalCalldata, encodedMessage);
+        assertTrue(ism.isVerified(encodedMessage));
+    }
+
+    /* ============ ISM.verify ============ */
+
+    function test_verify_revertWhen_invalidMetadata() public virtual {
+        bool isValid;
+        try ism.verify(new bytes(0), encodedMessage) returns (bool _isValid) {
+            isValid = _isValid;
+        } catch {}
+        assertFalse(isValid);
```

Comment:
why not expectRevert?

Reply:
Not all external bridges revert if invalid (like 5164), they just return false, this is a general catch-all.

Reply:
fixed by overridding case by case